syntax = "proto3";

package com.embedd;
option go_package = "backend/core/proto;proto";

enum FileType {
    // ORCHESTRATOR SEND TO CONNECTOR:
    UNKNOWN = 0;
    PDF = 1;
    XPS = 2;
    TXT = 3;
    DOCX = 4;
    DOC = 5;
    XLSX = 6;
    XLS = 7;
    PPTX = 8;
    PPT = 9;
    HWPX = 10;
    EPUB = 11;
    MOBI = 12;
    FB2 = 13;
    CBZ = 14;
    MD = 15;
    // ORCHESTRATOR SEND TO SEMANTIC
    URL = 16;
    YT = 17;

    // ORCHESTRATOR SEND TO MEDIA
    MP4 = 18;
    MP3 = 19;
    MPEG = 20;
    MPGA = 21;
    M4A = 22;
    WAV = 23;
    WEBM = 24;
    MOV = 25;

  // add all supported file that in another document
  // check what with Google docs
};

message SemanticData {
  // This is the url where the file is located.
  // Based on the chunking type it will be a WEB URL (HTML type)
  // Will be an S3/MINIO link with a proper authentication in case of a file
  string url = 1;
  bool url_recursive = 2;
  string site_map = 3;
  bool search_for_sitemap = 4;
  int64  document_id = 5;







  int64  connector_id = 6;
  FileType file_type = 7;
  string collection_name = 8;
  string model_name = 9;
  int32 model_dimension = 10;
}
